Semantics.fs

```diff
@@ -88,6 +99,82 @@ let frame svars tvars expr =
         | Some i -> Some (name, ty, (fun x -> (Intermediate(i, x)))))
     |> Seq.map (fun (name, ty, ctor) -> frameVar ctor (withType ty name))
 
+let paramToMExpr : TypedVar -> SMExpr =
```

Comment:
Just a heads up that these will need to be changed when we do arrays, it seems like there should be some way to abstract over the actual type being used here?